Userland/Libraries/LibC/semaphore.cpp

```diff
@@ -95,7 +99,8 @@ sem_t* sem_open(char const* name, int flags, ...)
     pthread_mutex_lock(&s_sem_mutex);
     ScopeGuard unlock_guard = [] { pthread_mutex_unlock(&s_sem_mutex); };
 
-    int fd = open(path.characters(), O_RDWR | O_CLOEXEC | flags, mode);
+    // sem_name_to_path guarantees a null terminator.
+    int fd = open(path.bytes_as_string_view().characters_without_null_termination(), O_RDWR | O_CLOEXEC | flags, mode);
```

Comment:
I wonder if we should be using some type of wrapper here like what is done in Core::System::open to avoid needing to enforce the above guarantee.  Especially since the serenity syscall itself does not expect null termination